sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -640,6 +644,7 @@ case class StructsToJson(
   lazy val rowSchema = child.dataType match {
     case st: StructType => st
     case ArrayType(st: StructType, _) => st
+    case MapType(_: DataType, _: DataType, _) => child.dataType
```

Comment:
case mt: MapType => mt

Reply:
We can add a todo: `//TODO: support ArrayType(MapType)`, if we don't plan to add the support in this PR.